lib/spack/docs/build_systems/intelpackage.rst

```diff
@@ -0,0 +1,994 @@
+.. _intelpackage:
+
+.. contents::
+
+
+--------------------------
+Intel packages in Spack
+--------------------------
+
+Spack can work with several software development products offered by Intel.
+Some of these are available under no-cost terms, others require a paid license.
+All share the same basic steps for configuration, installation, and, where
+applicable, license management. The relevant Spack Python class is:
+
+.. code-block:: python
+
+  IntelPackage
+
+Spack handles several interaction routes with Intel tools, like it does for any
+other package:
+
+.. _`route 1`:
+
+1. Accept system-provided tools in Spack after you declare them as *external packages*.
+
+.. _`route 2`:
+
+2. Install the products for you as *internal packages* in Spack.
+
+.. _`route 3`:
+
+3. *Use* the packages, regardless of installation route, to install what we'll
+   call *client packages* for you, this being Spack's primary purpose.
+
+An auxiliary route follows from `route 2`_, as it would for most Spack
+packages, namely:
+
+.. _`route 4`:
+
+4. Make Spack-installed Intel tools available outside of Spack for ad-hoc use,
+   typically through Spack-managed modulefiles.
+
+This document covers routes 1 through 3.
+
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Packages under no-cost license
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Intel's standalone performance library products, notably MKL and MPI, are
+available for use under a `simplified license
+<https://software.intel.com/en-us/license/intel-simplified-software-license>`_
+since 2017 [fn1]_. They are packaged in Spack as:
+
+* ``intel-mkl`` – Math Kernel Library (linear algebra and FFT),
+* ``intel-mpi`` – The Intel-MPI implementation (derived from MPICH),
+* ``intel-ipp`` – Primitives for image-, signal-, and data-processing,
+* ``intel-daal`` – Machine learning and data analytics.
+
+Some earlier versions of these library products were released under a paid
+license, which for these versions must be available, like for compilers
+discussed next, at installation time of the products and during compilation of
+client packages.
+
+The library packages can be used both with and without Intel compilers.
+The latter offer options to simplify linking (sometimes considerably),
+but Spack always uses fairly explicit linkage instead.
+
+
+^^^^^^^^^^^^^^^^^^
+Licensed packages
+^^^^^^^^^^^^^^^^^^
+
+Intel's core software development products that provide compilers, analyzers,
+and optimizers do require a paid license.  In Spack, they are packaged as:
+
+* ``intel-parallel-studio`` – the entire suite of compilers and libraries,
+* ``intel`` – a subset containing just the compilers and the Intel-MPI runtime [fn2]_.
```

Comment:
Absolutely, and this turns out to be a portal into the abyss. To illustrate, if I may: https://mobile.twitter.com/BradyHaran/status/1001153108037259264 . See #8632 and related.

I tried maintaining the package selections as they were, for the most part.

Quarantining MPI runtimes from nominally non-MPI-containing `studio` editions would be possible for Spack-internal installations only; we cannot, however, affect `mpirun` etc. in Spack-external packages. I have a feeling this could get quite hairy.